pilot/pkg/model/push_context.go

```diff
@@ -1640,6 +1640,12 @@ func (ps *PushContext) SetDestinationRules(configs []config.Config) {
 
 	for i := range configs {
 		rule := configs[i].Spec.(*networking.DestinationRule)
+		// warning log for the destination rule which belong to root namespace and with short name host, such as 'review'
+		if configs[i].Namespace == ps.Mesh.RootNamespace && rule.Host != "" && host.IsShortNameForHost(rule.Host) {
+			warningMSG := "Namespace/mesh-level DestinationRule [%q] is belonged to root namespace with short name host [%q], " +
+				"recommend to provide the whole host, such as [%q].default.svc.cluster.local."
+			log.Warnf(warningMSG, configs[i].Name, rule.Host, rule.Host)
```

Comment:
This is not a great place to warn users. The person who applies Destinationrule is unlikely to be looking at logs in istiod, and this will log very often. it is better suited to place the warning in `pkg/config/validation/validation.go` which is exposed to use when they `kubectl apply`

Reply:
Great suggestion, it's the right place to warn users, there is a new submit. Thanks @howardjohn ! ^_^